src/Microsoft.DotNet.Build.Tasks/ValidateExactRestore.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json.Linq;
+using NuGet.Versioning;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ValidateExactRestore : Task
+    {
+        [Required]
+        public ITaskItem[] ProjectLockJsons { get; set; }
+
+        private HashSet<string> _errors = new HashSet<string>();
+
+        public override bool Execute()
+        {
+            foreach (var projectLockItem in ProjectLockJsons)
+            {
+                ValidateLockFile(projectLockItem.ItemSpec);
+            }
+            return !Log.HasLoggedErrors;
+        }
+
+        private void ValidateLockFile(string lockFilePath)
+        {
+            var lockfile = VisitProjectDependencies.ReadJsonFile(lockFilePath);
+
+            var lockedDependencyVersions = lockfile["libraries"]
+                .Children<JProperty>()
+                .Select(p => p.Name.Split('/'))
+                .ToLookup(libParts => libParts[0], libParts => NuGetVersion.Parse(libParts[1]));
+
+            // Look at dependencies for all frameworks.
+            var requestedDependencies = lockfile["projectFileDependencyGroups"]
+                .Children<JProperty>()
+                .SelectMany(group => group.Value.Value<JArray>().Values<string>());
+
+            foreach (var dependency in requestedDependencies)
+            {
+                string[] dependencyParts = dependency.Split(' ');
```

Comment:
FYI: here is the source for this string https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.LibraryModel/LibraryRange.cs#L66